Task6/AbstractFactory/PaperBookAbstractFactory.cs

```diff
@@ -0,0 +1,145 @@
+                                    }
+                                    else if (subItems.Length == 1)
+                                    {
+                                        author = new Author(subItems[0], string.Empty, string.Empty);
+                                    }
+                                    else if (regex2.IsMatch(subItems[1]))
+                                    {
+                                        author = new Author(subItems[0], string.Empty, string.Empty);
+                                    }
+                                    else
+                                    {
+                                        author = new Author(subItems[1], subItems[0], string.Empty);
+                                    }
+
+                                    authorList.Add(author);
+                                }
+                                break;
+                            case "related-external-id":
+                                string patternIsbn = @"isbn:\d+";
+
+                                Regex regexIsbn = new Regex(patternIsbn);
+                                MatchCollection matches = regexIsbn.Matches(fields[i]);
+
+                                foreach (Match match in matches)
+                                {
+                                    isbns.Add(match.Value.Substring(5));
+                                }
+                                break;
+                            case "publisher":
+                                string[] publishersArray = fields[i].Split(';');
+
+                                foreach (string item in publishersArray)
+                                {
+                                    if (!_uniquePublishers.Contains(item.Trim()))
+                                    {
+                                        _uniquePublishers.Add(item.Trim());
+                                    }
+
+                                    publishers.Add(item.Trim());
+                                }
+                                break;
+                        }
+                    }
+                    if (isbns.Count == 0)
+                    {
+                        isbns.Add(isbnNull.ToString());
+                        isbnNull++;
+                        isbnIsNull = true;
+                    }
+                    
+                    PaperBook paperBook = new PaperBook(title, publicationDate, authorList, isbns, publishers, isbnIsNull);
+                    if (catalog.CatalogEntrys.ContainsKey(paperBook.Isbns.FirstOrDefault()))
+                    {
+                        catalog.Add(paperBook.Isbns[2], paperBook);
+                    }
+                    else
+                    {
+                        catalog.Add(paperBook.Isbns.FirstOrDefault(), paperBook);
+                    }
+                }
+            }
+
+            return catalog;
+        }
+
```

Comment:
Very bad idea, you suppose that CreateCatalog() will be called before the call of CreatePressReleaseItems(), this is not truth. As interface method it can be called separately, and you will return nulll. Parse  case "publisher" section separately

Reply:
Fixed